SDAF - OpenQA API directly on SUT (#20722)
os-autoinst/os-autoinst-distri-opensuse: tests/sles4sap/redirection_tests/README.md

# Redirection test modules

## Overview

Test modules in this directory are intended for testing scenarios with remote SUT (System under test).  
This means worker VM is serving only as an SSH jumphost to SUT.

```| OpenQA instance| --> | QEMU Worker | --> Open SSH terminal --> | System under test|```   

This is achieved using console redirection located in `lib/sles4sap/console_redirection.pm`.  

> Test modules should be independent of a deployment solution and should not really contain code related to it. 

Reviewer: ```suggestion
Test modules is independent of a particular deployment solution and does not contain code related to it. 
```

Author: Thanks, just fixed it.

Technically,

> you should be able to run test with two (or more)

Reviewer 2: Is the line separation here intentional?

        },
        hanadb_b => {
            ip_address => '192.168.1.4',
            ssh_user   => 'hanaadmin'
        }
    },
    db_ase => {
        asedb => {
            ip_address => '192.168.1.5',
            ssh_user   => 'hanaadmin'
        }
    },    
    nw_pas   => {
        nw_pas => {
            ip_address => '192.168.1.6',
            ssh_user   => 'hanaadmin'
        }
    },
    nw_aas   => {
        nw_aas_01 => {
            ip_address => '192.168.1.7',
            ssh_user   => 'hanaadmin'
        },
        nw_aas_02 => {
            ip_address => '192.168.1.8',
            ssh_user   => 'hanaadmin'
        }
    },
    nw_ascs  => {
        nw_ascs => {
            ip_address => '192.168.1.9',
            ssh_user   => 'hanaadmin'
        }
    },
    nw_ers   => {
        nw_ers => {
            ip_address => '192.168.1.10',
            ssh_user   => 'hanaadmin'
        }
    }
};
```

Please keep information in the data structure only relevant to console redirection.  
If a test module needs additional information (data about sap instances),
create a new required structure and document it.
Try to make this structure generic,

> not based on a specific document outputted from a specific deployment solution like a tfvars file.

Reviewer 2: Line formatting is also weird here.
